hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -132,9 +132,12 @@ type HostedClusterReconciler struct {
 	// 2) The OCP version being deployed is the latest version supported by Hypershift
 	HypershiftOperatorImage string
 
-	// releaseProvider looks up the OCP version for the release images in HostedClusters
+	// ReleaseProvider looks up the OCP version for the release images in HostedClusters
 	ReleaseProvider releaseinfo.ProviderWithRegistryOverrides
 
+	// ReleaseVersion is the OCP release image semver
+	ReleaseVersion semver.Version
```

Comment:
@hasueki I don't think this belongs in the reconciler. The reconciler will be reconciling HostedClusters of potentially many different release versions.

Reply:
Ah gotcha, I was afraid that was the case. I went back and forth on where we can put this info. It seems like we don't want to do a registry lookup every time either.. Would it be ok if we add a new field in `HC.Status.Version.Desired{}`? Maybe `HC.Status.Version.Desired.Version`?
https://github.com/openshift/hypershift/blob/d1aaa9b3dee008976cf00657583fe96975ee7bb8/api/v1beta1/hostedcluster_types.go#L1264-L1270

Reply:
The registry lookup is not bad, we use a cache that should result in a quick operation. 
https://github.com/openshift/hypershift/blob/d1aaa9b3dee008976cf00657583fe96975ee7bb8/hypershift-operator/main.go#L233

Reply:
Is the caching behavior different for images with tags vs digests? I think there were few convos on that for using digests for HO/CPO images. Is that the same case for this release image lookup? We use release images that may look like: `ocp-release:4.9.51-x86_64`

Reply:
Ah I see, it looks like it uses a different cache for release images vs the scenario I was thinking about:
- https://github.com/openshift/hypershift/blob/d1aaa9b3dee008976cf00657583fe96975ee7bb8/support/releaseinfo/cached_provider.go#L40
- https://github.com/openshift/hypershift/blob/d1aaa9b3dee008976cf00657583fe96975ee7bb8/support/util/imagemetadata.go#L50

I will adjust to do the release image lookup instead.